### PR TITLE
#15282 Fix single \r to \r\n before query execution

### LINKED ISSUE #15282: SQL execution with Inline double-dash comment
DBeaver 21.3.3, Windows 10

Working with SQLServer database, the execution of a query with a double-dash comment results in an execution error, while with MySQL, using the same query, the execution ends correclty
By opening the executed query tab, it looks like that for the SQLServer version the new lines are not kept

**MySQL**
image

**SQLServer**
image

### PR DESCRIPTION
**Pull request by E1izabeth**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLUtils.java [hunk 1]
```diff
@@ -740,23 +740,25 @@ public static String fixLineFeeds(String sql) {
             return sql;
         }
         boolean hasFixes = false;
-        char[] fixed = sql.toCharArray();
-        for (int i = 0; i < fixed.length; i++) {
-            if (fixed[i] == '\r') {
-                if (i > 0 && fixed[i - 1] == '\n') {
+        char[] initial = sql.toCharArray();
+        StringBuilder fixed = new StringBuilder();
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLUtils.java]:
```suggestion
        StringBuilder fixed = new StringBuilder(initial.length);
```

**Review by ShadelessFox — COMMENTED** [review]
What do you think about adding a unit test for this method?

**Comment by E1izabeth** [comment]
It's a good idea.

**Review by ShadelessFox — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**